fplus-lib/src/core/mod.rs

```diff
@@ -4044,85 +4044,71 @@ _The initial issue can be edited in order to solve the request of the verifier.
-                client_id
-            ))
-        })?;
-
-        let is_first = last_allocation.request_type == "First";
+        let is_first = application_file.get_active_allocation_request_type()? == "First";
         if is_first {
             self.remove_first_pending_allocation(&application_file)
                 .await?;
         } else {
             self.remove_pending_refill(&app_model.pr_number).await?;
         }
-        self.issue_updates_after_removing_pending_allocation(
-            &application_file.issue_number,
-            is_first,
-        )
-        .await?;
-        Ok(())
-    }
 
-    async fn issue_updates_after_removing_pending_allocation(
-        &self,
-        issue_number: &str,
-        is_first: bool,
-    ) -> Result<(), LDNError> {
         let comment = format!(
             "Last pending allocation reverted for an application `{}`.",
             &self.application_id
         );
-        let issue_number = issue_number.parse::<u64>().map_err(|e| {
-            LDNError::New(format!(
-                "Parse issue number: {} to u64 failed. {}",
-                issue_number, e
-            ))
-        })?;
-        self.github
-            .add_comment_to_issue(issue_number, &comment)
-            .await
-            .map_err(|e| {
-                LDNError::New(format!(
-                    "Error adding comment to issue {} /// {}",
-                    issue_number, e
-                ))
-            })?;
 
         let app_state = if is_first {
             AppState::Submitted.as_str()
         } else {
             AppState::Granted.as_str()
         };
-        self.github
-            .replace_issue_labels(issue_number, &[app_state.into()])
-            .await
-            .map_err(|e| {
-                LDNError::New(format!(
-                    "Error adding labels to issue {} /// {}",
-                    issue_number, e
-                ))
-            })?;
+
+        self.issue_updates(&application_file.issue_number, &comment, app_state)
+            .await?;
+        Ok(())
+    }
+
+    pub async fn revert_to_ready_to_sign(
+        &self,
+        client_id: &str,
+        owner: &str,
+        repo: &str,
+    ) -> Result<(), LDNError> {
+        let app_model =
+            Self::get_application_model(client_id.into(), owner.into(), repo.into()).await?;
+
+        let application_file =
+            Self::get_application_file_with_active_allocation(&app_model).await?;
+
+        if application_file.lifecycle.state != AppState::StartSignDatacap {
+            return Err(LDNError::Load(format!(
+                "Application state is {:?}. Expected StartSignDatacap",
+                application_file.lifecycle.state
+            )));
+        }
+
+        self.remove_signers_from_active_request(&application_file)
+            .await?;
+
+        let comment = format!(
+            "Pending allocation reverted to ReadyToSign state for an application: {:?}.",
```

Comment:
This is too technical. Why was it reverted? What is `ReadyToSign`?
Maybe `Allocation transaction failed on chain, please try again.`?